includes/sanitizers/class-amp-layout-sanitizer.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Class AMP_Layout_Sanitizer
+ *
+ * @since 1.5.0
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Layout_Sanitizer
+ *
+ * @since 1.5.0
+ */
+class AMP_Layout_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Sanitize any element that has the `layout` or `data-amp-layout` attribute.
+	 */
+	public function sanitize() {
+		$xpath = new DOMXPath( $this->dom );
+		$nodes = $xpath->query( '//*[ @layout or @data-amp-layout ]' );
```

Comment:
Given that `layout` is validated in `AMP_Tag_And_Attribute_Sanitizer`, this query needs only to be for `data-amp-layout` to copy it over to `layout`.

Reply:
So invalid layouts that can be fixed should be ignored?

Reply:
Done in https://github.com/ampproject/amp-wp/pull/3728/commits/27ac54f7e43e6df22686c28cca9bff7e5269bda1.